clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java

```diff
@@ -65,16 +67,59 @@ protected void maybeThrowAuthFailure(Node node) {
         networkClientDelegate.maybeThrowAuthFailure(node);
     }
 
+    /**
+     * Request that a fetch request be issued to the cluster to pull down the next batch of records.
+     *
+     * <p/>
+     *
+     * The returned {@link CompletableFuture} is {@link CompletableFuture#complete(Object) completed} when the
+     * fetch request(s) have been created and enqueued into the network client's outgoing send buffer.
+     * It is <em>not completed</em> when the network client has received the data.
+     *
+     * @return Future for which the caller can wait to ensure that the requests have been enqueued
```

Comment:
as above? the completion only tells that the request has been generated right?

Reply:
Updated the comments. Thanks!